src/librustdoc/html/static/js/main.js

```diff
@@ -1003,36 +1003,67 @@ function preLoadCss(cssUrl) {
         window.register_type_impls(window.pending_type_impls);
     }
 
+    // Draw a convenient sidebar of known crates if we have a listing
     function addSidebarCrates() {
-        // @ts-expect-error
         if (!window.ALL_CRATES) {
             return;
         }
         const sidebarElems = document.getElementById("rustdoc-modnav");
         if (!sidebarElems) {
             return;
         }
-        // Draw a convenient sidebar of known crates if we have a listing
-        const h3 = document.createElement("h3");
-        h3.innerHTML = "Crates";
-        const ul = document.createElement("ul");
-        ul.className = "block crate";
 
-        // @ts-expect-error
-        for (const crate of window.ALL_CRATES) {
-            const link = document.createElement("a");
-            link.href = window.rootPath + crate + "/index.html";
-            link.textContent = crate;
-
-            const li = document.createElement("li");
-            if (window.rootPath !== "./" && crate === window.currentCrate) {
-                li.className = "current";
+        const allCratesHeading = document.createElement("h3");
+        allCratesHeading.textContent = "Crates";
+
+        const allCratesSection = document.createElement("section");
+
+        // window.ALL_CRATES is in unsorted array-of-structs format; reorganize so crates with the
+        // same heading are grouped.
+        const cratesGroupedByHeading = new Map();
+        for (const entry of window.ALL_CRATES) {
+            const heading = entry.h;
+            const crateName = entry.c;
+            let group = cratesGroupedByHeading.get(heading);
+            if (group === undefined) {
+                group = [];
+                cratesGroupedByHeading.set(heading, group);
             }
-            li.appendChild(link);
-            ul.appendChild(li);
+            group.push(crateName);
+        }
+        const headings = Array.from(cratesGroupedByHeading.keys());
+        headings.sort();
+
+        // Generate HTML for grouped crates.
+        for (const headingText of headings) {
+            // Empty string denotes a group with no named heading.
+            if (headingText !== "") {
+                const crateCategoryHeading = document.createElement("h4");
```

Comment:
To fix this, make `allCratesHeading` an `h2` and make `crateCategoryHeading` an `h3`.

I'm not sure why an `h3` is used here. It's not a sibling with other `h3`s, like "Crate Items" and "Sections". It's a sibling with the crate name heading itself (in your screenshot, the "Crates" list is a sibling with "all_is_cubes_mesh"). It's also analogous with the "In Module all_is_cubes_mesh::whatever" headline, which is an `h2` also).

Reply:
Done.

Reply:
Here is what the updated headings look like. Still no hierarchy in the amount of whitespace, but it is at least not wildly inconsistent.

<img width="258" alt="image" src="https://github.com/user-attachments/assets/515e3964-a293-4301-b878-5b2394855ffd" />